src/statements.md

```diff
@@ -54,17 +54,36 @@ fn outer() {
 > **<sup>Syntax</sup>**\
 > _LetStatement_ :\
 > &nbsp;&nbsp; [_OuterAttribute_]<sup>\*</sup> `let` [_PatternNoTopAlt_]
->     ( `:` [_Type_] )<sup>?</sup> (`=` [_Expression_] )<sup>?</sup> `;`
+>     ( `:` [_Type_] )<sup>?</sup> ( | `=` [_Expression_] | `=` [_Expression_]
+>     `else` [_BlockExpression_]) `;`
```

Comment:
Do you know how to do the note in source code?

Reply:
(also yes the leading | was to indicate the epsilon case, but I have changed it to your suggestion)

Reply:
Oh, sorry, I should have included the source from the comment above:

```
> **<sup>Syntax</sup>**\
> … [_Expression_] [†](#let-else-restriction) …
> 
> <span id="let-else-restriction">† When an `else` block is specified, the _Expression_ must not be a [_LazyBooleanExpression_](expressions/operator-expression.md#lazy-boolean-operators) or end with a `}`.</span>
```

Just make it a link, and use a `<span>` to set the target.

Reply:
Thanks! I've tried out your code, it causes an error in CI:

> error in ../src/statements.md (line 57): broken Reference link (reference `†`)
some style checks failed

Apparently pulldown-cmark does not parse the inline html.

I have tried using a footnote, but that shows a <sup>1</sup>, which in this context can be very easily misunderstood as a repetition:
![Screenshot_20220924_031548](https://user-images.githubusercontent.com/8872119/192074031-572317e2-d44f-4681-9f42-4ceeb6db9804.png)

However, it is possible to hide the number and put a † on its stead. I've pushed a commit to do that.

Reply:
I would prefer to not change the behavior of footnotes. Those are used in other places of the reference.

I would also prefer to avoid using them in this situation. This isn't really a footnote, but more of an attempt to stuff a bunch of information in a small amount of space.

I tried the suggestion I posted above, and the `style-check` and the linkchecker both passed.  It should look exactly like:

```md
> **<sup>Syntax</sup>**\
> _LetStatement_ :\
> &nbsp;&nbsp; [_OuterAttribute_]<sup>\*</sup> `let` [_PatternNoTopAlt_]
>     ( `:` [_Type_] )<sup>?</sup> (`=` [_Expression_] [†](#let-else-restriction)
>     ( `else` [_BlockExpression_]) <sup>?</sup> ) <sup>?</sup> `;`
>
> <span id="let-else-restriction">† When an `else` block is specified, the _Expression_ must not be a [_LazyBooleanExpression_](expressions/operator-expr.md#lazy-boolean-operators) or end with a `}`.</span>
```

Reply:
@ehuss regarding the CI failure, please see [this](https://github.com/rust-lang/reference/actions/runs/3116465400/jobs/5054319422) CI run, ran from [this](https://github.com/rust-lang/reference/pull/1156/commits/3eacf0029c5298c481fca9db1e1adf5d313ed293) commit. If you click on the prev button you will get to the commit that has precisely the state that you suggest.

```
    Compiling style-check v0.1.0 (/home/runner/work/reference/reference/style-check)
    Finished dev [unoptimized + debuginfo] target(s) in 37.63s
     Running `target/debug/style-check ../src`
error in ../src/statements.md (line 57): broken Reference link (reference `†`)
some style checks failed
```

```markdown
> &nbsp;&nbsp; [_OuterAttribute_]<sup>\*</sup> `let` [_PatternNoTopAlt_]
>     ( `:` [_Type_] )<sup>?</sup> (`=` [_Expression_] )<sup>?</sup> `;`
>     ( `:` [_Type_] )<sup>?</sup> (`=` [_Expression_][†](#let-else-restriction)
>     ( `else` [_BlockExpression_]) <sup>?</sup> ) <sup>?</sup> `;`
A *`let` statement* introduces a new set of [variables], given by an
<span id="let-else-restriction">† When an `else` block is specified, the
irrefutable [pattern]. The pattern is followed optionally by a type
_Expression_ must not be a [_LazyBooleanExpression_], or end with a `}`.</span>
```

I can try it again, but I doubt it will be any different.

Reply:
That's because in that commit you have:

```
[_Expression_][†](#let-else-restriction)
```

The text I have above has a space:

```
[_Expression_] [†](#let-else-restriction)
```

Without the space, markdown thinks the `[†]` is a reference link for the preceding `[_Expression_]`.

Reply:
Ohhhh... I see... Will adjust it right away.